sphinx/tutorial/rose/furthertopics/optional-configurations.rst

```diff
@@ -0,0 +1,181 @@
+more than one in the same directory) are named ``rose-app-NAME.conf``
+where ``NAME`` is some kind of useful id.
+
+
+New Optional Configurations
+---------------------------
+
+Create an optional configuration file called ``rose-app-chocofudge.conf``
+in your new ``opt/`` subdirectory, with the following content:
+
+.. code-block:: rose
+
+   [env]
+   FLAVOUR=chocolate
+   TOPPING=fudge
+
+We can now reference this through an option to ``rose task-run`` - alter
+the ``script`` of the ``order_icecream`` task (in the ``suite.rc`` file)
+to read:
+
+.. code-block:: cylc
+
+   script = rose task-run --opt-conf-key=chocofudge
+
+This will now override the ``order_icecream`` app's ``rose-app.conf``
+file with the contents of the ``rose-app-chocofudge.conf`` file at runtime.
+
+We can add as many ``--opt-conf-key`` options to ``rose task-run`` as we
+like - not just the one. This would apply each optional configuration in
+first-to-last order.
+
+Try running the suite - it should apply the optional configuration correctly.
+
+
+$ROSE_APP_OPT_CONF_KEYS
+-----------------------
+
+We could also specify the optional configurations to use by passing in an
+environment variable ``$ROSE_APP_OPT_CONF_KEYS``, a space delimited list
+of optional configuration names to use.
+
+Try using it by replacing the order_icecream task configuration with:
+
+.. code-block:: cylc
+
+   [[order_icecream]]
+       script = rose task-run
+       [[[environment]]]
+           ROSE_APP_OPT_CONF_KEYS = chocofudge
+
+If you re-run the suite, this should do the same job as the ``--opt-conf-key``
+method we tried before.
+
+
+Default Optional Configurations
+-------------------------------
+
+Optional configurations can be switched on by default by specifying their
+names via a top-level opts setting in the configuration file.
+
+Put the following line at the top of the order_icecream ``rose-app.conf`` file:
+
+.. code-block:: rose
+
+   opts=chocofudge
+
+We now don't need to specify it at all for ``rose task-run`` - get rid of
+the environment variable in the ``suite.rc`` file, so that the
+``order_icecream`` task runtime configuration looks like this:
+
+.. code-block:: cylc
+
+   [[order_icecream]]
+       script = rose task-run
+
+
+Default Optional Configurations - Test
+--------------------------------------
+
+If you run the suite, it should have the same standard output for
+``order_icecream`` as before.
+
+Typically, because the opts setting is 'always-on', it's best to use the
+optional configurations to add settings rather than override them.
+
+
+Other Optional Configurations
+-----------------------------
+
+All Rose configurations can have optional configurations, not just
+application configurations. Suites can have optional configurations
+that override ``rose-suite.conf`` settings, controlled through
+``rose suite-run``. This takes the same ``--opt-conf-key`` option as
+``rose app-run`` and/or the environment variable ``$ROSE_SUITE_OPT_CONF_KEYS``.
```

Comment:
I think ``rose app-run`` & ``$ROSE_APP_OPT_CONF_KEYS``, or the equivalent pair for suites and not apps, were intended, i.e. it is best to have consistency in configuration type referenced.